src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -195,10 +269,25 @@ public static KNNQueryBuilder fromXContent(XContentParser parser) throws IOExcep
             }
         }
 
-        KNNQueryBuilder knnQueryBuilder = new KNNQueryBuilder(fieldName, ObjectsToFloats(vector), k, filter);
-        knnQueryBuilder.ignoreUnmapped(ignoreUnmapped);
-        knnQueryBuilder.queryName(queryName);
-        knnQueryBuilder.boost(boost);
+        if ((k != null && distance != null) || (k == null && distance == null)) {
+            throw new IllegalArgumentException("[" + NAME + "] requires either k or distance must be set");
+        }
+
+        KNNQueryBuilder knnQueryBuilder;
+        if (k != null) {
+            knnQueryBuilder = new KNNQueryBuilder(fieldName, ObjectsToFloats(vector)).k(k)
```

Comment:
move common fields to line 276 and set either k or distance inside if block

Reply:
Updated